### PR TITLE
Use repository.lower() when looking up existing commit and creating new one

### PR DESCRIPTION
**Pull request by ElenaHenderson**

### CODE REVIEW

File: conbench/entities/benchmark_result.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def create(data):
             repository = repository_to_url(data["github"]["repository"])
 
         # create if not exists
-        commit = Commit.first(sha=sha, repository=repository)
+        commit = Commit.first(sha=sha, repository={repository.lower() if repository else None})
```
Comment (austin3dickey) [inline conbench/entities/benchmark_result.py]: I don't think you need the set
```suggestion
        commit = Commit.first(
            sha=sha, repository=repository.lower() if repository else None
        )
```

**Review by austin3dickey — COMMENTED** [review]
Are there other places we need to wrap with `lower()`? For example should we store all the repos as lowercase on the commit entity?

File: conbench/entities/benchmark_result.py (hunk 1 above)
Comment (ElenaHenderson) [reply conbench/entities/benchmark_result.py]: Thank you! I have no idea what I was thinking.

**Comment by ElenaHenderson** [comment]
> Are there other places we need to wrap with `lower()`? For example should we store all the repos as lowercase on the commit entity?

Looked. Don't really see anything similar situations for other tables. I would also be careful to lower() anything unless we look for data on all conbench instances and make sure we have correct migrations.

**Comment by austin3dickey** [comment]
Hm, just thought of something. If a server is set up to track a repo with an uppercase letter, e.g. `https://github.com/my_org/MyRepo`, won't something strange happen?

1. someone posts a benchmark result for commit sha `"abc"`
2. a new Commit is created with `"sha": "abc", "repository": "https://github.com/my_org/MyRepo"`
3. someone posts another benchmark result on the same commit
4. this PR code doesn't find any matches for `sha="abc", repository="https://github.com/my_org/myrepo"`
5. it tries to create another Commit with `"sha": "abc", "repository": "https://github.com/my_org/MyRepo"`
6. that fails due to the unique index

That's why I was thinking we potentially have to store the repo as `.lower()`.

**Comment by ElenaHenderson** [comment]
> Hm, just thought of something. If a server is set up to track a repo with an uppercase letter, e.g. `https://github.com/my_org/MyRepo`, won't something strange happen?
> 
> 1. someone posts a benchmark result for commit sha `"abc"`
> 2. a new Commit is created with `"sha": "abc", "repository": "https://github.com/my_org/MyRepo"`
> 3. someone posts another benchmark result on the same commit
> 4. this PR code doesn't find any matches for `sha="abc", repository="https://github.com/my_org/myrepo"`
> 5. it tries to create another Commit with `"sha": "abc", "repository": "https://github.com/my_org/MyRepo"`
> 6. that fails due to the unique index
> 
> That's why I was thinking we potentially have to store the repo as `.lower()`.

You are right. I should have thought about this!

**Review by austin3dickey — APPROVED** [review]
Nice, looks good. One more thing; would you mind putting an uppercase repo in [these tests](https://github.com/conbench/conbench/blob/main/conbench/tests/entities/test_commit.py#L29)? Thanks!!

**Comment by ElenaHenderson** [comment]
> Nice, looks good. One more thing; would you mind putting an uppercase repo in [these tests](https://github.com/conbench/conbench/blob/main/conbench/tests/entities/test_commit.py#L29)? Thanks!!

Done.

**Pull request merged**